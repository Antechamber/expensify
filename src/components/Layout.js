import React from 'react'
import { Outlet, NavLink } from 'react-router-dom'

const Layout = () => (
    <div>
        <header>
            <h1>Expensify</h1>
            <ul>
                {[['Home', '/'], ['Add an expense', '/create']].map((page, i) => (
                    <li key={i}>
                        <NavLink to={page[1]} end>{page[0]}</NavLink>
                    </li>
                ))}
            </ul>
            <hr />
        </header>
        <Outlet />
    </div>
)

export default Layout