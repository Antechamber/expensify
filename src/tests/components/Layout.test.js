import React from 'react'
import { shallow } from 'enzyme'
import Layout from '../../components/Layout'

test('Should render layout component page', () => {
    const wrapper = shallow(<Layout />)
    expect(wrapper).toMatchSnapshot()
    // expect(wrapper.find('h1').text()).toBe('Expensify')
})
