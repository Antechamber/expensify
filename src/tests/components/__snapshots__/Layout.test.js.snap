// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render layout component page 1`] = `
<div>
  <header>
    <h1>
      Expensify
    </h1>
    <ul>
      <li
        key="0"
      >
        <[object Object]
          end={true}
          to="/"
        >
          Home
        </[object Object]>
      </li>
      <li
        key="1"
      >
        <[object Object]
          end={true}
          to="/create"
        >
          Add an expense
        </[object Object]>
      </li>
      <li
        key="2"
      >
        <[object Object]
          end={true}
          to="/help"
        >
          Help
        </[object Object]>
      </li>
    </ul>
    <hr />
  </header>
  <Outlet />
</div>
`;
